Installing-and-Configuring/OS/index.md

```diff
@@ -87,11 +85,11 @@ This model is vulnerable to brute force attacks but does not lock the data to th
 
 ### Security Dongle (aka smartcard aka security token)
 
-You may use a security dongle to manage the passphrase for the root LUKS partition.  This is also out of scope for heads even if using the same security dongle for verification such as in the [Pureboot](https://docs.puri.sm/PureBoot.html) model.  
+You may use a security dongle to manage the passphrase for the root LUKS partition.  This is also out of scope for heads even if using the same security dongle for verification.
 
-The first advantage to this approach is that the user only needs to type a PIN on boot rather than a passphrase.  Unlike the [heads TPM](#using-tpm-for-decryption-of-drives) model this model does not lock the data on disk to the hardware.  The hard drive may be moved to a different system and used with the security dongle and/or LUKS passphrase if it is still on the partition.  Also, the security dongle enforces limits which prevent brute force attacks.
+The first advantage to this approach is that the user only needs to type a PIN on boot rather than a passphrase.  Unlike the [heads TPM](#using-tpm-for-decryption-of-drives) model this model does not lock the data on disk to the hardware.  The hard drive may be moved to a different system and used with the security dongle and/or a LUKS passphrase.  Also, the security dongle enforces limits which prevent brute force attacks.
```

Comment:
@jtmoree-github-com 

> The first advantage to this approach is that the user only needs to type a PIN on boot rather than a passphrase. 

TPM based Disk Unlock Key passphrase doesn't force a specific length for the PIN/passphrase. It could actually be a PIN if the user wants so.  

> Unlike the [heads TPM](#using-tpm-for-decryption-of-drives) model this model does not lock the data on disk to the hardware.

TPM based Disk Unlock Key is not touching the Disk Recovery Key passphrase is kept intact into slot 0. 

kexec-seal-key [generates a key](https://github.com/osresearch/heads/blob/c3b0bd6ffbe816430dd41ef54e649af52ed1ff3b/initrd/bin/kexec-seal-key#L55-L60) (Disk Unlock Key) [that is injected in slot 1](https://github.com/osresearch/heads/blob/c3b0bd6ffbe816430dd41ef54e649af52ed1ff3b/initrd/bin/kexec-seal-key#L72-L76). A measurement of the resulting LUKS header [is then taken](https://github.com/osresearch/heads/blob/c3b0bd6ffbe816430dd41ef54e649af52ed1ff3b/initrd/bin/kexec-seal-key#L83-L85), which is then [sealted into TPM nv memory](https://github.com/osresearch/heads/blob/c3b0bd6ffbe816430dd41ef54e649af52ed1ff3b/initrd/bin/kexec-seal-key#L105-L118), with the Disk Unlock Key passphrase which can only be unlocked if the firmware measurements, LUKS header and Disk Unlock Key passphrase are coherent with the PCR values at the moment the sealing that secret into nv space, to be released when booting default boot option when Disk Unlock Key passphrase is not deactivated in board config, and when a TPM is present.

Also, since the TPM rate limits passphrase/PIN prompts, after the 3rd Disk Unlock Key passphrase/PIN attempt, Heads will propose to the user of booting the OS as is, where the OS will prompt for the Disk Recovery Key passphrase (slot 0 unlocking key).

The Disk Recovery Key can be used on a secondary computer to access the data of the encrypted partition.

Reply:
I'm rewriting these sections to incorporate your points and fix mine.

Do I understand correctly that we can effectively lock the disk to the hardware by erasing the passphrase in slot 0 when using heads to store the disk unlock passphrase in the TPM?
This would be similar using only 1 keyslot with a process locked to a specific private key on a security dongle.   In both cases the LUKS data is tied to keys that are available by using a particular hardware device.

Reply:
Forget the 'lock' language.  I am removing it all.  

Rewrote tpm and dongle sections added another section incorporating your notes.